spring-cloud-stream-tuple/src/main/java/org/springframework/cloud/stream/tuple/DefaultMutableTuple.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.tuple;
+
+import java.util.List;
+
+import org.springframework.core.convert.support.ConfigurableConversionService;
+
+/**
+ * Extension of {@link DefaultTuple} that also allows tuple {@link MutableTuple mutation}.
+ *
+ * @author Eric Bottard
+ */
+public class DefaultMutableTuple extends DefaultTuple implements MutableTuple {
+
+	public DefaultMutableTuple(List<String> names, List<Object> values, ConfigurableConversionService configurableConversionService) {
+		super(names, values, configurableConversionService);
+	}
+
+	@Override
+	public synchronized void setValue(int index, Object value) {
```

Comment:
Unless reads are also `synchronized`, I'm not sure that making these methods `synchronized` is worthwhile. IMO, either the entire class needs to be made thread safe, or the JavaDoc should explicitly state that it isn't thread safe.